src/main/java/org/truffleruby/language/arguments/KeywordArgumentsDescriptor.java

```diff
@@ -14,9 +14,14 @@
  * arguments were passed. The callee must handle that for now. */
 public class KeywordArgumentsDescriptor extends ArgumentsDescriptor {
 
-    public static final KeywordArgumentsDescriptor INSTANCE = new KeywordArgumentsDescriptor();
+    private final String[] keywords;
 
-    private KeywordArgumentsDescriptor() {
+    /** Use {@link KeywordArgumentsDescriptorManager} to get an instance. */
+    public KeywordArgumentsDescriptor(String[] keywords) {
```

Comment:
Make this package private to make it clearer it should go through KeywordArgumentsDescriptorManager